### PR TITLE
fix resource quota update for grace period

### PR DESCRIPTION
**Pull request by silenceshell**
**What type of PR is this?**

/kind bug

**What this PR does / why we need it**:

DeletionTimestamp means the time when this resource WILL be deleted.

When a pod is deleted, the controller set `pod.DeletionTimestamp` by addding the `pod.DeletionGracePeriodSeconds`, so there is no necessary to add `gracePeriod` here again.

And If the pod is deleted before gracePeriod, the controller will set `pod.DeletionTimestamp` to the current time and set `pod.DeletionGracePeriodSeconds` to 0. In this case, we should skip notifyUpdate and wait for the pod delete event.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

NA

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs
NONE
```

### CODE REVIEW

**Comment by yliaog** [comment]
first of all, please add a test to verify it, thanks.

File: pkg/quota/v1/evaluator/core/pods.go [hunk 1]
```diff
@@ -394,8 +394,7 @@ func QuotaV1Pod(pod *corev1.Pod, clock clock.Clock) bool {
 	if pod.DeletionTimestamp != nil && pod.DeletionGracePeriodSeconds != nil {
 		now := clock.Now()
 		deletionTime := pod.DeletionTimestamp.Time
-		gracePeriod := time.Duration(*pod.DeletionGracePeriodSeconds) * time.Second
-		if now.After(deletionTime.Add(gracePeriod)) {
+		if *pod.DeletionGracePeriodSeconds != 0 && now.After(deletionTime) {
```
Comment (derekwaynecarr) [inline pkg/quota/v1/evaluator/core/pods.go:397]: looking at:
https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/registry/rest/delete.go#L132

we can see that DeletionTimestamp is set based on current time + grace period as noted, so I agree it makes sense to not add the additional grace period to existing timestamp.

Given that, can't this check be simplified to just now.After(deletionTime) ?

**Review by derekwaynecarr — CHANGES_REQUESTED** [review]
I agree with the motivation for the change, I think we can simplify the code to just check for now.After(deletionTime) as we do not need to look at pod.DeletionGracePeriodSeconds at all.

File: pkg/quota/v1/evaluator/core/pods.go (hunk 1 above)
Comment (yliaog) [reply pkg/quota/v1/evaluator/core/pods.go:397]: i think actually it is not just for simplification, it is incorrect to have the condition *pod.DeletionGracePeriodSeconds != 0

Comment (silenceshell) [reply pkg/quota/v1/evaluator/core/pods.go]: Thanks for your review comments! @derekwaynecarr @yliaog  
Code updated.

**Comment by silenceshell** [comment]
ping @derekwaynecarr 
thanks!

**Pull request closed without merging**